crates/rooch-server/src/api/rooch_api.rs

```diff
@@ -2,68 +2,73 @@
 
-#[rpc(server, client)]
+#[open_rpc(namespace = "rooch")]
+#[rpc(server, client, namespace = "rooch")]
 pub trait RoochAPI {
     /// Send the signed transaction in bcs hex format
     /// This method does not block waiting for the transaction to be executed.
-    #[method(name = "rooch_sendRawTransaction")]
-    async fn send_raw_transaction(&self, tx_bcs_hex: StrView<Vec<u8>>) -> RpcResult<H256>;
+    #[method(name = "sendRawTransaction")]
+    async fn send_raw_transaction(&self, tx_bcs_hex: StrView<Vec<u8>>) -> RpcResult<RoochH256View>;
 
     /// Send the signed transaction in bcs hex format
     /// This method blocks waiting for the transaction to be executed.
-    #[method(name = "rooch_executeRawTransaction")]
+    #[method(name = "executeRawTransaction")]
     async fn execute_raw_transaction(
         &self,
         tx_bcs_hex: StrView<Vec<u8>>,
     ) -> RpcResult<ExecuteTransactionResponseView>;
 
     /// Execute a read-only function call
     /// The function do not change the state of Application
-    #[method(name = "rooch_executeViewFunction")]
+    #[method(name = "executeViewFunction")]
     async fn execute_view_function(
         &self,
         function_call: FunctionCallView,
     ) -> RpcResult<Vec<AnnotatedFunctionReturnValueView>>;
 
     /// Get the states by access_path
-    #[method(name = "rooch_getStates")]
-    async fn get_states(&self, access_path: AccessPath) -> RpcResult<Vec<Option<StateView>>>;
+    #[method(name = "getStates")]
+    async fn get_states(&self, access_path: AccessPathView) -> RpcResult<Vec<Option<StateView>>>;
 
     /// Get the annotated states by access_path
     /// The annotated states include the decoded move value of the state
-    #[method(name = "rooch_getAnnotatedStates")]
+    #[method(name = "getAnnotatedStates")]
     async fn get_annotated_states(
         &self,
-        access_path: AccessPath,
+        access_path: AccessPathView,
     ) -> RpcResult<Vec<Option<AnnotatedStateView>>>;
 
     /// Get the events by event handle id
-    #[method(name = "rooch_getEventsByEventHandle")]
+    #[method(name = "getEventsByEventHandle")]
     async fn get_events_by_event_handle(
         &self,
         event_handle_type: StructTagView,
         cursor: Option<u64>,
         limit: Option<u64>,
-    ) -> RpcResult<EventPage>;
+    ) -> RpcResult<EventPageView>;
 
     /// Get the events by event filter
-    #[method(name = "rooch_getEvents")]
-    async fn get_events(&self, filter: EventFilter) -> RpcResult<Vec<Option<AnnotatedEventView>>>;
+    #[method(name = "getEvents")]
+    async fn get_events(
+        &self,
+        filter: EventFilterView,
+    ) -> RpcResult<Vec<Option<AnnotatedEventView>>>;
 
-    #[method(name = "rooch_getTransactionByHash")]
-    async fn get_transaction_by_hash(&self, hash: H256) -> RpcResult<Option<TransactionView>>;
+    #[method(name = "getTransactionByHash")]
+    async fn get_transaction_by_hash(
+        &self,
+        hash: RoochH256View,
```

Comment:
Do all non-primitive types must  be View types or are they optional?

Reply:
View just for better json display. We can discuss on the next dev meeting.